lumicks/pylake/kymotracker/kymotracker.py

```diff
@@ -376,14 +391,20 @@ def refine_tracks_centroid(tracks, track_width=None):
     if not tracks:
         return KymoTrackGroup([])
 
+    minimum_width = 3 * tracks._kymo.pixelsize[0]
     if track_width is None:
-        track_width = _default_track_widths[tracks._kymo._calibration.unit]
-
-    if track_width <= 0:
-        # Must be positive otherwise refinement fails
-        raise ValueError("track_width should be larger than zero")
+        track_width = max(_default_track_widths[tracks._kymo._calibration.unit], minimum_width)
+
+    # We lower the minimum bound slightly to ensure that setting it to the minimum bound exactly
+    # becomes a valid value.
+    if track_width < np.nextafter(minimum_width, 0):
+        # The window should at least contain 3 pixels otherwise refinement fails.
+        raise ValueError(
+            f"track_width should at least be 3 pixels ({3 * tracks._kymo.pixelsize[0]:.2f} "
+            f"[{tracks._kymo._calibration.unit}])"
+        )
```

Comment:
considering this is duplicated from `track_greedy` and it's not in a loop, should this maybe be pulled out to a separate `validate_parameters()` function? Not a huge deal, but at least there'd be one source of validation and we could add other checks in there in the future if need be

Reply:
Yeah, good call.